### PR TITLE
Rust subscribe example

### PR DESCRIPTION
**Pull request by carlmartus**
Subscribe and print to signals specified.

### CODE REVIEW

**Comment by carlmartus** [comment]
Should be on top of #37.

- [ ] Rebase to `main` when #37 is merged

**Comment by carlmartus** [comment]
- [ ] Use new method of refering to signals: `Namespace.Frame.Signal`

**Comment by JohanKarlbergg** [comment]
> * [ ]  Use new method of refering to signals: `Namespace.Frame.Signal`

It is <namespace>:<signal name> reference
https://github.com/remotivelabs/remotivelabs-samples/pull/20/files

File: rust/remotivelabs-broker/examples/subscribe.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+/// Rust subscribe example
+#[derive(Parser, Debug)]
+#[command(about)]
+struct Args {
+    /// Namespace
+    #[arg(short, long, required = true)]
+    namespace: String,
+
+    #[arg(short, long, required = true)]
+    signal: Vec<String>,
```
Comment (JohanKarlbergg) [inline rust/remotivelabs-broker/examples/subscribe.rs:23]: One argument which is namespace:signal

File: rust/remotivelabs-broker/examples/subscribe.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args = Args::parse();
+
+    let channel = Channel::from_static("http://localhost:50051")
```
Comment (jankronquist) [inline rust/remotivelabs-broker/examples/subscribe.rs:30]: lets make broker URL a command line arg also

**Comment by jankronquist** [comment]
does it work using a cloud broker? if not then maybe we can ask for pull requests? :)

File: rust/remotivelabs-broker/README.md [hunk 3]
```diff
@@ -19,3 +21,11 @@ Now the sample `sub_client` is running and waiting for signals. To produce signa
     cargo run --example pub_client
 
 Now follow the prompt instruction and input numbers. The numbers will be received and printed by the first program.
+
+### Subscribe
```
Comment (jankronquist) [inline rust/remotivelabs-broker/README.md:25]: put this example first!

File: rust/remotivelabs-broker/README.md [hunk 4]
```diff
@@ -6,7 +6,9 @@ gRPC bindings are generated from [this source](../../proto/) when compiling the
 
 This project is contributed by _Niclas Lind_ (@niclaslind).
 
-## Publish / Subscribe example
+## Examples
+
+### Publish / Subscribe example
 See `examples/pub_client.rs` and `examples/sub_client.rs` for a sample on how to use the library.
 
 The samples work together and will connect to a _RemotiveBroker_ running on `localhost`. First start the subscription sample:
```
Comment (jankronquist) [inline rust/remotivelabs-broker/README.md:14]: mention that this example is using the "virtual" interface over the RemotiveBroker

File: rust/remotivelabs-broker/examples/subscribe.rs (hunk 2 above)
Comment (carlmartus) [reply rust/remotivelabs-broker/examples/subscribe.rs]: Added. Also supports for X-API key.

File: rust/remotivelabs-broker/README.md (hunk 4 above)
Comment (carlmartus) [reply rust/remotivelabs-broker/README.md]: This is updated in #37

File: rust/remotivelabs-broker/examples/subscribe.rs (hunk 1 above)
Comment (carlmartus) [reply rust/remotivelabs-broker/examples/subscribe.rs]: Added

**Comment by carlmartus** [comment]
> does it work using a cloud broker? if not then maybe we can ask for pull requests? :)

Yes. As long as the user provides the X-API key. Which is argument `-x`. This is in PR #37.

File: rust/remotivelabs-broker/README.md (hunk 3 above)
Comment (carlmartus) [reply rust/remotivelabs-broker/README.md]: Moved

**Review by jankronquist — APPROVED** [review]

**Pull request merged**